src/combinatorics/hall.lean

```diff
@@ -0,0 +1,341 @@
+import data.fintype.basic
+
+open finset
+
+universe u
+
+variables {α β : Type u} [fintype α] [fintype β]
+variables (r : α → β → Prop) [∀ a, decidable_pred (r a)]
+
+/--
+The image of relation `r` from `α` in `β`.
+-/
+def image_rel (A : finset α) : finset β := univ.filter (λ b, ∃ a ∈ A, r a b)
```

Comment:
Perhaps this belongs in `data/rel` instead?

Reply:
Replaced it with `rel.image`